lib/helpers/menu_options.rb

```diff
@@ -1,26 +1,81 @@
 require_relative 'object_creation'
 require_relative 'preserve_data'
 
-def list_all(arr, label)
-  data = load_data(arr, label)
-
+def list_all(arr, label, data)
+  puts ''
   puts "List of all #{label}"
-  print 'title | author | genre | source'
-  data.each_with_index do |item, index|
-    print (index + 1).to_s
-    print "#{item.label.title} | "
-    print "#{item.author.first_name} "
-    print "#{item.author.last_name} | "
+  puts 'id | title | author | genre | source'
+  puts '____________________________________'
+  arr.each do |item|
+    add_missing_data(item, data)
+    print "#{item.id} | "
+    # print "#{item.label.title} | "
+    # print "#{item.author.name} "
+    # print "#{item.author.last_name} | "
     print "#{item.genre.name} | "
-    print "#{item.source.name} \n"
+    # print "#{item.source.name} \n"
   end
   run
 end
 
-def load_data(arr, file)
-  data = recover_data(file)
-  data = arr if data.nil?
-  data
+def loop_method(arr, id)
+  arr.map do |e|
+    e if e.id == id
+  end
+end
+
+def add_missing_data(item, data)
+  return unless item.genre.is_a?(Integer)
+
+  genreobj = loop_method(data[:genres_list], item.genre)
+  item.add_genre(genreobj[0])
+
+  # labelobj = loop_method(data[:labels_list], item.label)
+  # item.add_label(labelobj[0])
+
+  # authorobj = loop_method(data[:authors_list], item.author)
+  # item.add_author(authorobj[0])
+
+  # sourceobj = loop_method(data[:sources_list], item.source)
+  # item.add_source(sourceobj[0])
+end
+
+def list_all_genre_source(arr, label)
+  puts ''
+  puts "List of all #{label}"
+  puts 'id | name'
+  puts '___________'
+  arr.each do |item|
+    print "#{item.id} | "
+    print "#{item.name} \n"
+  end
+  run
+end
+
+def list_all_label(arr, label)
+  puts ''
+  puts "List of all #{label}"
+  puts 'id | title | color'
+  puts '___________'
+  arr.each do |item|
+    print "#{item.id} | "
+    print "#{item.title} | "
+    print "#{item.color} \n"
+  end
+  run
+end
+
+def list_all_author(arr, label)
+  puts ''
+  puts "List of all #{label}"
+  puts 'id | name | last name'
+  puts '___________'
+  arr.each do |item|
+    print "#{item.id} | "
+    print "#{item.name} | "
+    print "#{item.last_name} \n"
+  end
+  run
```

Comment:
For these functions, I think there's no need to pass a ```label``` argument, now that we have different functions to list the elements, we can delete the ```#{label}``` from the ```print``` statement and instead put the label directly inside the quotations like:
``` puts 'List of all colors'```

Reply:
Great, thanks I Will change it and merge it